### PR TITLE
Updating cognify pipeline documentation

### PR DESCRIPTION
**Pull request by alekszievr**
Updated the cognify pipeline documentation to match current code. Got rid of outdated docs.

## Summary by CodeRabbit

- **Documentation**
	- Introduced a comprehensive overview of pipeline functionality within the Cognee framework, detailing task organization and execution for data processing.
	- Added new sections and descriptions for key functions and classes related to pipeline management, including `cognee.cognify`, `run_tasks`, and `run_tasks_parallel`.
	- Removed outdated documentation that previously outlined task organization and execution in pipelines.

### CODE REVIEW

File: docs/pipelines.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+This is the main pipeline currently implemented in cognee. It is designed to process data in a structured way and populate the graph and vector stores with the results.
+
+
+This function is the entry point for processing datasets. It handles dataset retrieval, user authorization, and manages the execution of a pipeline of tasks that process documents.
+
+### Parameters
+
+- `dataset: Union[str, list[str]] = None`: A string or list of dataset names to be processed.
+- `user: User = None`: The user requesting the processing. If not provided, the default user is retrieved.
```
[the remark this answers is not in the document]
Comment (alekszievr) [reply docs/pipelines.md]: fixed

File: docs/pipelines.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+```python
+pipeline = run_tasks(tasks, documents)
+async for result in pipeline:
+    print(result)
+```
+
+## Main pipeline: [cognee.cognify](https://github.com/topoteretes/cognee/blob/168cb5d1bf1964b5b0c645b2f3d8638d84554fda/cognee/api/v1/cognify/cognify_v2.py#L38)
+
+This is the main pipeline currently implemented in cognee. It is designed to process data in a structured way and populate the graph and vector stores with the results.
```
Comment (hajdul88) [inline docs/pipelines.md:15]: Here maybe it's a bit confusing to use the word "results". To me it means the final results that are generated by the LLM at the end (either search or QA). I would just delete the "with the results" part. Not critical tho.

**Review by hajdul88 — APPROVED** [review]
I think, it is a great improvement of the documentation.

File: docs/pipelines.md (hunk 2 above)
Comment (alekszievr) [reply docs/pipelines.md]: good point, fixed

**Pull request merged**